src/System.Windows.Forms.Primitives/src/Interop/WinFormsComWrappers.IDataObjectVtbl.cs

```diff
@@ -0,0 +1,185 @@
+                vtblRaw[0] = fpQueryInterface;
+                vtblRaw[1] = fpAddRef;
+                vtblRaw[2] = fpRelease;
+                vtblRaw[3] = (IntPtr)(delegate* unmanaged<IntPtr, FORMATETC*, STGMEDIUM_Raw*, HRESULT>)&GetData;
+                vtblRaw[4] = (IntPtr)(delegate* unmanaged<IntPtr, FORMATETC*, STGMEDIUM_Raw*, HRESULT>)&GetDataHere;
+                vtblRaw[5] = (IntPtr)(delegate* unmanaged<IntPtr, FORMATETC*, HRESULT>)&QueryGetData;
+                vtblRaw[6] = (IntPtr)(delegate* unmanaged<IntPtr, FORMATETC*, FORMATETC*, HRESULT>)&GetCanonicalFormatEtc;
+                vtblRaw[7] = (IntPtr)(delegate* unmanaged<IntPtr, FORMATETC*, STGMEDIUM_Raw*, int, HRESULT>)&SetData;
+                vtblRaw[8] = (IntPtr)(delegate* unmanaged<IntPtr, DATADIR, IntPtr*, HRESULT>)&EnumFormatEtc;
+                vtblRaw[9] = (IntPtr)(delegate* unmanaged<IntPtr, FORMATETC*, ADVF, IntPtr, int*, HRESULT>)&DAdvise;
+                vtblRaw[10] = (IntPtr)(delegate* unmanaged<IntPtr, int, HRESULT>)&DUnadvise;
+                vtblRaw[11] = (IntPtr)(delegate* unmanaged<IntPtr, IntPtr*, HRESULT>)&EnumDAdvise;
+
+                return (IntPtr)vtblRaw;
+            }
+
+            [UnmanagedCallersOnly]
+            private static HRESULT GetData(IntPtr thisPtr, FORMATETC* format, STGMEDIUM_Raw* pMedium)
+            {
+                var inst = ComInterfaceDispatch.GetInstance<IDataObject>((ComInterfaceDispatch*)thisPtr);
+                try
+                {
+                    inst.GetData(ref *format, out var medium);
+                    pMedium->pUnkForRelease = medium.pUnkForRelease == null ? IntPtr.Zero : Marshal.GetIUnknownForObject(medium.pUnkForRelease);
+                    pMedium->tymed = medium.tymed;
+                    pMedium->unionmember = medium.unionmember;
+                    return HRESULT.S_OK;
+                }
+                catch (Exception ex)
+                {
+                    return (HRESULT)ex.HResult;
+                }
+            }
+
+            [UnmanagedCallersOnly]
+            private static unsafe HRESULT GetDataHere(IntPtr thisPtr, FORMATETC* format, STGMEDIUM_Raw* pMedium)
+            {
+                var inst = ComInterfaceDispatch.GetInstance<IDataObject>((ComInterfaceDispatch*)thisPtr);
+                try
+                {
+                    STGMEDIUM medium = new();
+                    medium.pUnkForRelease = Marshal.GetObjectForIUnknown(pMedium->pUnkForRelease);
+                    medium.tymed = pMedium->tymed;
+                    medium.unionmember = pMedium->unionmember;
+                    inst.GetDataHere(ref *format, ref medium);
+                    pMedium->pUnkForRelease = medium.pUnkForRelease == null ? IntPtr.Zero : Marshal.GetIUnknownForObject(medium.pUnkForRelease);
+                    pMedium->tymed = medium.tymed;
+                    pMedium->unionmember = medium.unionmember;
+                    return HRESULT.S_OK;
```

Comment:
The code would benefit from reformatting and having few empty lines to improve readability
```suggestion
                    STGMEDIUM medium = new()
                    {
                        pUnkForRelease = Marshal.GetObjectForIUnknown(pMedium->pUnkForRelease);
                        tymed = pMedium->tymed;
                        unionmember = pMedium->unionmember;
                    };

                    instance.GetDataHere(ref *format, ref medium);

                    pMedium->pUnkForRelease = medium.pUnkForRelease is null ? IntPtr.Zero : Marshal.GetIUnknownForObject(medium.pUnkForRelease);
                    pMedium->tymed = medium.tymed;
                    pMedium->unionmember = medium.unionmember;

                    return HRESULT.S_OK;
```

ditto other places